src/sov/mod.rs

```diff
@@ -0,0 +1,295 @@
+                                format!("Unable to parse SOV settings from file '{}'", settings.sov_source)
+                            ).into()
+                        );
+                    };
+            sov
+        } else {
+            // Presume it's an inline SOV settings string.
+            let Ok(sov) = serde_json::from_str::<SOVResponse>(&settings.sov_source) else {
+                return Err(
+                    HandlerErrorKind::Internal(
+                        format!("Could not parse SOV settings inline: {:?}", &settings.sov_source)
+                    ).into()
+                );
+            };
+            sov
+        };
+
+        Ok(SOVManager {
+            source_url: None,
+            refresh_rate: Duration::from_secs(settings.sov_refresh_rate_secs),
+            encoded_sov: Some(
+                // Unwrapping is safe here since the parsing above ensures it's a valid `SOVResponse`.
+                base64::engine::general_purpose::STANDARD_NO_PAD
+                    .encode(serde_json::to_string(&sov).unwrap()),
+            ),
+            last_response: Some(LastResponse {
+                response: sov,
+                updated: Utc::now(),
+            }),
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use serde_json::json;
+
+    use super::*;
+    use crate::web::test::get_test_settings;
+
+    const MOCK_SOV: &str = "eyJuYW1lIjoiU09WLTIwMjMwNTE4MjE1MzE2IiwiYWxsb2NhdGlv\
+    bnMiOlt7InBvc2l0aW9uIjoxLCJhbGxvY2F0aW9uIjpbeyJwYXJ0bmV\
+    yIjoiYW1wIiwicGVyY2VudGFnZSI6MTAwfV19LHsicG9zaXRpb24iOj\
+    IsImFsbG9jYXRpb24iOlt7InBhcnRuZXIiOiJhbXAiLCJwZXJjZW50Y\
+    WdlIjo4OH0seyJwYXJ0bmVyIjoibW96LXNhbGVzIiwicGVyY2VudGFn\
+    ZSI6MTJ9XX1dfQ";
+
+    #[test]
+    #[should_panic(expected = "Unable to parse SOV URL 'gs://bad^^url'")]
+    fn test_bad_gcs_url() {
+        let mut settings = Settings {
+            sov_source: "gs://bad^^url".to_owned(),
+
+            ..get_test_settings()
+        };
+        let _bad_url = HandlerResult::<SOVManager>::from(&mut settings).unwrap();
+    }
+    #[test]
+    #[should_panic(
+        expected = "Unable to parse SOV settings from file './test-engineering/contract-tests/volumes/contile/adm_settings.json'"
+    )]
+    fn test_bad_path() {
+        let mut settings = Settings {
+            sov_source: "./test-engineering/contract-tests/volumes/contile/adm_settings.json"
+                .to_owned(),
+
+            ..get_test_settings()
+        };
+        HandlerResult::<SOVManager>::from(&mut settings).unwrap();
+    }
+
+    #[test]
+    fn test_valid_path() {
+        let mut settings = Settings {
+            sov_source: "./test-engineering/contract-tests/volumes/contile/sov_settings.json"
+                .to_owned(),
+            ..get_test_settings()
+        };
+
+        let sov_manager = HandlerResult::<SOVManager>::from(&mut settings);
+        assert_eq!(sov_manager.unwrap().encoded_sov, Some(MOCK_SOV.to_owned()));
```

Comment:
Nit: doesn't really matter for testing, but here is how you can get an `Option<&str>` from an `Option<String>`, which saves a memory allocation.
```suggestion
        assert_eq!(sov_manager.unwrap().encoded_sov.as_deref(), Some(MOCK_SOV));
```

Reply:
oooh